terraform/azure/modules/vnet/vnet.tf

```diff
@@ -0,0 +1,50 @@
+#To create resource group
+resource "azurerm_resource_group" "rg" {
+  name     = var.resource_group_name
+  location = var.location
+}
+
+#To create virtual network
+resource "azurerm_virtual_network" "Vnet" {
+  name                = var.vnetwork_name
+  address_space       = ["10.0.0.0/16"]
+  location            = var.location
+  resource_group_name = var.resource_group_name
+}
+
+#To create subnets
+resource "azurerm_subnet" "subnets" {
+  count                = length(var.subnet_names)
+  name                 = var.subnet_names[count.index]
+  resource_group_name  = var.resource_group_name
+  virtual_network_name = var.virtual_network_name
+  address_prefixes     = ["10.0.2.0/24"]
+}
+
+resource "azurerm_subnet" "aks_subnet" {
```

Comment:
is there a point in keeping each of these as separate subnets? if not, I think its better to use the resource on line 16 based on the count of subnet_names itself?